juju/juju
# [JUJU-1222] Update LXD dependency to the latest 5.3 release

LXD has some changes to behaviour since our venerable version of the dependency.

Here we update it to the latest 5.2 version. Additional changes include:
- Changes to certificate post requests for the new payload structure.
- Updates to the logging proxy to handle LXD's use of [Logrus](https://github.com/Sirupsen/logrus).
- Updates to error type detection to use the LXD helpers.

## QA steps

- Bootstrap to lxd/localhost.
- Include `lxd=TRACE` in the controller config `logging-level`.
- Deploy some workloads and observe the output in controller logs.

## Documentation changes

None.

## Bug reference

N/A
